### ISSUE TITLE
Uncaught TypeError: klass.lastIndexOf is not a function

### ISSUE THREAD

**Issue by francisrod01** [grizzthedj/react-bootstrap-dropdown-menu#4]
Since I updated `react` to `^16.4.2` I've been a console error when I'll select some option in another dropdown.

**The screenshot:**

![image](https://user-images.githubusercontent.com/4691768/45153578-de7b9d80-b1aa-11e8-87a9-f47e729d91be.png)

```
Uncaught TypeError: klass.lastIndexOf is not a function
    at DropdownMenu.js:189
```

The `DropdownMenu.js:189` line mentions the `klass.lastIndexOf` property. See below:

```js
    key: 'componentWillMount',
    value: function componentWillMount() {
      var TRIGGER_CLASS = this.TRIGGER_CLASS;
      var MENUITEMS_DIV = this.MENUITEMS_DIV;
      var CARAT_CLASS = this.CARAT_CLASS;

      window.addEventListener("click", function (e) {
        var klass = e.target.className;
        var carat = document.getElementById(CARAT_CLASS);

        if (klass !== MENUITEMS_DIV + " show" && klass !== TRIGGER_CLASS && !klass.lastIndexOf("glyphicon", 0) == 0) {
          var menuItemDiv = document.getElementById(MENUITEMS_DIV);

          if (menuItemDiv) {
            menuItemDiv.classList.remove('show');

            if (carat) {
              carat.className = "glyphicon glyphicon-triangle-bottom";
            }
          }
        }
      });
    }
```

**Comment by grizzthedj** [comment]
@francisrod01 - Thanks for reporting this. Will have a look in the next day or two.

**Comment by grizzthedj** [comment]
@francisrod01 - My apologies for not getting to this earlier, but it has been really busy for me of late. 
 
Could you send me the rendered HTML for the other dropdown in question? I'm having a hard time reproducing.

**Comment by francisrod01** [comment]
Sure! It is the `react-select` with `grouped` implementation using a `multiple` property on `<Select>` component.

Here an example: https://react-select.com/home

I create an example on CodeSandbox for it :)

https://codesandbox.io/s/0qjmoq707p

Regards!

**Comment by grizzthedj** [comment]
@francisrod01 Sorry, this was long overdue. I completely forgot about this. This should be fixed in the latest version 1.1.21. I was not able to reproduce it, but I suspect that there is some element that has a numeric `className`.  Report back here if there are still issues. Again, my apologies for taking so long to fix this!

**Comment by RossVertizan** [comment]
@grizzthedj I had the same issue.  In my case the problem was that I had an event rippling up the event stack and getting caught by your component.  The correct way to deal with this is to use event.stopPropagation at the point where the event should be handled.  However, it might take a while to find all places where events are not getting caught, so I implemented the following in your code:

`window.addEventListener("click", function(e) {
      // console.log("DropdownMenu: componentWillMount: event: ", e);
      // console.log("DropdownMenu: componentWillMount: event.target: ", e.target);
      // Only process this event if it is actually coming from this component
      if (e.target.id.search("react_bs_dd") > -1) {
        // console.log("DropdownMenu: componentWillMount: Is the dropdown");
        const klass = e.target.className;
        const carat = document.getElementById(CARAT_CLASS);

        if (klass !== MENUITEMS_DIV + " show" && klass !== TRIGGER_CLASS && !klass.lastIndexOf("glyphicon", 0) == 0) {
          var menuItemDiv = document.getElementById(MENUITEMS_DIV);

          if (menuItemDiv) {
            menuItemDiv.classList.remove('show');

            if (carat) {
              carat.className = "glyphicon glyphicon-triangle-bottom";
            }
          }
        }
      }
    })`

[Sorry for the mess with the formatting of the code]

I added the `if (e.target.id.search("react_bs_dd") > -1) {`, so the handler is only executed if it is coming from your object, which traps it for all cases.

With your new version where you have added a check for a numeric class name I still see an error in the console.

Regards,
Ross

**Comment by grizzthedj** [comment]
@RossVertizan - I added a custom data attribute to the menu items div, and am conditionally handling the event based on that. This was published in 1.1.23. Let me know if this resolves the issue.

**Comment by RossVertizan** [comment]
@grizzthedj  - Sadly that did not help.  I have to admit to being a little confused by what you are doing here.  As far as I can see you get the menuItemDiv by searching the DOM for the "MENUITEMS_DIV" and then check that the menuItemDiv.dataset.reactbsdditems equals the MENUITEMS_DIV.  To be honest I'm not sure why it wouldn't because you have specifically searched for the div that has these settings.

Here is some debugging I added:

`window.addEventListener("click", function(e) {
      const klass = e.target.className;
      const carat = document.getElementById(CARAT_CLASS);
      const menuItemDiv = document.getElementById(MENUITEMS_DIV);

      console.log("DropdownMenu: menuItemDiv: ", menuItemDiv);
      console.log("DropdownMenu: menuItemDiv.dataset.reactbsdditems: ", menuItemDiv.dataset.reactbsdditems);
      console.log("DropdownMenu: MENUITEMS_DIV: ", MENUITEMS_DIV);
      console.log("DropdownMenu: event: ", e);
      if (menuItemDiv && menuItemDiv.dataset.reactbsdditems === MENUITEMS_DIV) {
        if (klass !== MENUITEMS_DIV + " show" && klass !== TRIGGER_CLASS && !klass.lastIndexOf("glyphicon", 0) == 0) {
          menuItemDiv.classList.remove('show');

          if (carat) {
            carat.className = "glyphicon glyphicon-triangle-bottom";
          }
        }
      }
    });`

[Sorry still not sure why the code formatting doesn't work]

... and here is some output from this:

`DropdownMenu: menuItemDiv:  <div data-reactbsdditems=​"__react_bs_dd_menuItems_1" id=​"__react_bs_dd_menuItems_1" class=​"__react_bs_dd_menuItems_1" style=​"display:​ none;​ position:​ absolute;​ left:​ 0px;​ z-index:​ 9999;​ background-color:​ rgb(249, 249, 249)​;​ min-width:​ 180px;​ padding:​ 12px;​ overflow:​ auto;​ box-shadow:​ rgba(0, 0, 0, 0.2)​ 0px 8px 16px 0px;​">​…​</div>​`

`DropdownMenu: menuItemDiv.dataset.reactbsdditems:  __react_bs_dd_menuItems_1`

`DropdownMenu: MENUITEMS_DIV:  __react_bs_dd_menuItems_1`

`DropdownMenu: event:  MouseEvent {isTrusted: true, screenX: -946, screenY: 431, clientX: 705, clientY: 230, …}altKey: falsebubbles: truebutton: 0buttons: 0cancelBubble: falsecancelable: trueclientX: 705clientY: 230composed: truectrlKey: falsecurrentTarget: nulldefaultPrevented: falsedetail: 1eventPhase: 0fromElement: nullisTrusted: truelayerX: 49889layerY: 49989metaKey: falsemovementX: 0movementY: 0offsetX: 49857offsetY: 49998pageX: 705pageY: 230path: (24) [svg, div#vitaq_diagram_canvas.react-js-diagrams-canvas, div#vitaq-diagram-tabs-pane-Test2.fade.tab-pane.active.in, div.tab-content, div#vitaq-diagram-tabs.tab_group, div.tabs_react_root, div#canvas.canvas, div.pane_div, div.Pane.horizontal.Pane1., div.SplitPane.horizontal., div.Pane.vertical.Pane1., div.SplitPane.vertical., div.Pane.vertical.Pane2., div.SplitPane.vertical., div.Pane.horizontal.Pane2., div.SplitPane.horizontal.disabled, div.panes_container, section.frontend-container, section.frontend-container-full, div#root, body, html, document, Window]relatedTarget: nullreturnValue: truescreenX: -946screenY: 431shiftKey: falsesourceCapabilities: InputDeviceCapabilities {firesTouchEvents: false}srcElement: svgtarget: svgtimeStamp: 37867.324999999255toElement: svgtype: "click"view: Window {0: Window, window: Window, self: Window, document: document, name: "", location: Location, …}which: 1x: 705y: 230__proto__: MouseEvent`

`Uncaught TypeError: klass.lastIndexOf is not a function
    at DropdownMenu.js:264`


I think you need to compare a property on the event to see if it matches the div - I don't think you are doing that at the moment.

Thanks
